addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/ApplianceGroheOndusHandler.java

```diff
@@ -0,0 +1,247 @@
+import org.eclipse.smarthome.core.types.State;
+import org.grohe.ondus.api.OndusService;
+import org.grohe.ondus.api.model.Appliance;
+import org.grohe.ondus.api.model.ApplianceCommand;
+import org.grohe.ondus.api.model.ApplianceData;
+import org.grohe.ondus.api.model.ApplianceData.Measurement;
+import org.grohe.ondus.api.model.Location;
+import org.grohe.ondus.api.model.Room;
+import org.openhab.binding.groheondus.internal.ApplianceGroheOndusConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florian Schmidt and Arne Wohlert - Initial contribution
+ */
+@NonNullByDefault
+public class ApplianceGroheOndusHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ApplianceGroheOndusHandler.class);
+
+    private @Nullable ApplianceGroheOndusConfiguration config;
+
+    public ApplianceGroheOndusHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ApplianceGroheOndusConfiguration.class);
+        scheduler.submit(() -> {
+            updateChannels();
+            updateStatus(ThingStatus.ONLINE);
+        });
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        super.channelLinked(channelUID);
+
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+        updateChannel(channelUID, appliance, getLastMeasurement(appliance));
+    }
+
+    public void updateChannels() {
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+
+        Measurement measurement = getLastMeasurement(appliance);
+        getThing().getChannels().forEach(channel -> updateChannel(channel.getUID(), appliance, measurement));
+
+        int pollingInterval = getPollingInterval(appliance);
+        scheduler.schedule(this::updateChannels, pollingInterval, TimeUnit.SECONDS);
+    }
+
+    private int getPollingInterval(Appliance appliance) {
+        if (config.pollingInterval > 0) {
+            return config.pollingInterval;
+        }
+        return appliance.getConfig().getMeasurementTransmissionIntervall();
+    }
+
+    private void updateChannel(ChannelUID channel, Appliance appliance, Measurement measurement) {
+        String channelName = channel.getIdWithoutGroup();
```

Comment:
the better variable name would be `channelId`.

Reply:
Right, done! :)